src/java/org/apache/cassandra/schema/CompressionParams.java

```diff
@@ -127,11 +119,50 @@ public static CompressionParams fromMap(Map<String, String> opts)
         return cp;
     }
 
+    public static CompressionParams fromParameterizedClass(ParameterizedClass parameterizedClass)
+    {
+        Map<String, String> options = new HashMap<>(parameterizedClass.parameters);
+        options.put( CLASS, parameterizedClass.class_name);
+
+        String sstableCompressionClass;
+
+
+        if (!removeEnabled(options))
+        {
+            sstableCompressionClass = null;
+
+            if (!options.isEmpty())
+                throw new ConfigurationException(format("If the '%s' option is set to false no other options must be specified", ENABLED));
```

Comment:
this is interesting. I am curious why there is such condition in place. What is wrong with having parameters in config if the feature is not enabled? This is common pattern in cassandra.yaml, no?

Reply:
this is normal for schema config classes.